lib/Controller/PageController.php

```diff
@@ -34,27 +34,41 @@
 use OCP\IRequest;
 use OCP\AppFramework\Http\TemplateResponse;
 use OCP\AppFramework\Controller;
+use OCA\Deck\Db\CardMapper;
+use OCP\IURLGenerator;
+use \OCP\AppFramework\Http\RedirectResponse;
+use OCA\Deck\Db\Acl;
+use OCA\Deck\Service\CardService;
 
 class PageController extends Controller {
 	private $permissionService;
 	private $initialState;
 	private $configService;
 	private $eventDispatcher;
+	protected $cardMapper;
+	private $urlGenerator;
+	private $cardService;
 
 	public function __construct(
 		$AppName,
 		IRequest $request,
 		PermissionService $permissionService,
 		IInitialStateService $initialStateService,
 		ConfigService $configService,
-		IEventDispatcher $eventDispatcher
+		IEventDispatcher $eventDispatcher,
+		CardMapper $cardMapper,
+		IURLGenerator $urlGenerator,
+		CardService $cardService
 		) {
 		parent::__construct($AppName, $request);
 
 		$this->permissionService = $permissionService;
 		$this->initialState = $initialStateService;
 		$this->configService = $configService;
 		$this->eventDispatcher = $eventDispatcher;
+		$this->cardMapper = $cardMapper;
+		$this->urlGenerator = $urlGenerator;
```

Comment:
Those are not needed anymore now, as the url generation has been moved to the CardService 😉